internal/engine/netx/errorx/errorx.go

```diff
@@ -320,3 +389,11 @@ func toOperationString(err error, operation string) string {
 	}
 	return operation
 }
+
+func isCertificateError(alert uint8) bool {
+	return (alert == TLSBadCertificate ||
```

Comment:
Where does this logic come from? Is it based on empirical experience or is it mirroring some existing code?

Reply:
This is based on my understanding of the error condition described here: https://datatracker.ietf.org/doc/html/rfc8446#section-6.2

Reply:
I would add a comment saying that, then!